src/oatpp/core/data/stream/Stream.cpp

```diff
@@ -284,6 +284,82 @@ StreamType DefaultInitializedContext::getStreamType() const {
   return m_streamType;
 }
 
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// BufferedInputStream
+
+v_io_size BufferedInputStream::flushBufferToStream(stream::WriteCallback *writeCallback) {
+  return writeBufferToStream(writeCallback, availableToRead());
+}
+
+async::CoroutineStarter BufferedInputStream::flushBufferToStreamAsync(const std::shared_ptr<data::stream::WriteCallback> &writeCallback) {
+  return writeBufferToStreamAsync(writeCallback, availableToRead());
+}
+
+v_io_size BufferedInputStream::writeBufferToStream(stream::WriteCallback *writeCallback, v_buff_size count) {
+  auto ioBuffer = buffer::IOBuffer::createShared();
+  count = std::min(count, (v_buff_size)availableToRead());
+  transfer(this, writeCallback, count, ioBuffer->getData(), ioBuffer->getSize());
+  return count;
+}
+
+async::CoroutineStarter BufferedInputStream::writeBufferToStreamAsync(const std::shared_ptr<data::stream::WriteCallback>& writeCallback, v_buff_size count) {
+  auto ioBuffer = buffer::IOBuffer::createShared();
+  return transferAsync(this, writeCallback, count, ioBuffer);
+}
+
+v_io_size BufferedInputStream::writeBufferToStream(stream::WriteCallback *writeCallback,
+                                                   v_buff_size count,
+                                                   v_buff_size readOffset) {
+  auto ioBuffer = buffer::IOBuffer::createShared();
+  if (readOffset > availableToRead()) {
+    commitReadOffset(availableToRead());
+    return 0;
+  }
+  commitReadOffset(readOffset);
+  count = std::min(count, (v_buff_size)availableToRead());
+  transfer(this, writeCallback, count, ioBuffer->getData(), ioBuffer->getSize());
+  return count;
+}
+
+async::CoroutineStarter BufferedInputStream::writeBufferToStreamAsync(const std::shared_ptr<data::stream::WriteCallback> &stream,
+                                                                      v_buff_size count,
+                                                                      v_buff_size readOffset) {
+  auto ioBuffer = buffer::IOBuffer::createShared();
+  if (readOffset > availableToRead()) {
+    class NothingToDo : public async::Coroutine<NothingToDo> {
```

Comment:
`NothingToDo` coroutine is not needed.
` return nullptr;` will do

Reply:
TIL. Thats good to know!